src/components/OnboardingComponents/SemanticSearch/ArticleDialog.tsx

```diff
@@ -0,0 +1,210 @@
+// SPDX-FileCopyrightText: 2024 German Aerospace Center (DLR)
+// SPDX-License-Identifier: Apache-2.0
+
+import React, {useState, useRef, useEffect, useCallback} from 'react';
+import {
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  IconButton,
+  Typography,
+  Box,
+  CircularProgress,
+  Tooltip,
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import ArrowBackIosNewIcon from '@mui/icons-material/ArrowBackIosNew';
+import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
+import ZoomInIcon from '@mui/icons-material/ZoomIn';
+import ZoomOutIcon from '@mui/icons-material/ZoomOut';
+import {SearchResult} from 'types/semanticSearch';
+import {Document, Page, pdfjs} from 'react-pdf';
+import 'react-pdf/dist/Page/AnnotationLayer.css';
+import 'react-pdf/dist/Page/TextLayer.css';
+
+pdfjs.GlobalWorkerOptions.workerSrc = `//unpkg.com/pdfjs-dist@${pdfjs.version}/build/pdf.worker.min.mjs`;
+
+interface ArticleDialogProps {
+  open: boolean;
+  onClose: () => void;
+  article: SearchResult | null;
+}
+
+/**
+ * A dialog component to display the full content of a selected search result article.
+ */
+export default function ArticleDialog({open, onClose, article}: ArticleDialogProps): JSX.Element {
+  const [numPages, setNumPages] = useState<number>(0);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [zoom, setZoom] = useState(1);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
```

Comment:
Please add comments to your useEffects to summarize their purpose/function.

```suggestion
  // Effect to register ResizeObserver to adjust Dialog to window resizes.
  useEffect(() => {
```

Reply:
Added.